test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -450,10 +450,26 @@ session_simulator::set_durable_timestamp(uint64_t ts)
     _durable_ts = ts;
 }
 
-void
-session_simulator::set_prepare_timestamp(uint64_t ts)
+int
+session_simulator::set_prepare_timestamp(uint64_t prepare_ts)
 {
-    _prepare_ts = ts;
+    timestamp_manager *ts_manager = &timestamp_manager::get_timestamp_manager();
+    WT_SIM_RET(ts_manager->validate_prepare_timestamp(this, prepare_ts));
+
+    /*
+     * If the given timestamp is earlier than the oldest timestamp then round the prepare timestamp
+     * to oldest timestamp if round prepared is true.
+     */
+    if (_ts_round_prepared) {
+        connection_simulator *conn = &connection_simulator::get_connection();
+        uint64_t oldest_ts = conn->get_oldest_ts();
```

Comment:
Since the `conn` object is just used once, can we do instead:
`connection_simulator::get_connection()-> get_oldest_ts()`

Reply:
Sean chose to do it that way as he found it more readable. Also, for consistency sake let's keep it the way it is.